torch/autograd/variable.py

```diff
@@ -582,10 +582,10 @@ def dot(self, other):
     def _addcop(self, op, args):
         if len(args) == 3:
             # scale, tensor1, tensor2
-            return op(args[0])(self, *args[1:])
+            return op.apply(self, args[1], args[2], args[0])
         else:
             # tensor1, tensor2
-            return op()(self, *args)
+            return op.apply(self, *args)
 
     def addcmul(self, *args):
```

Comment:
Can you also expose inplace versions of these function? (Just modify `_addcop` to accept a third `inplace` arg)? It seems that we forgot about them. Please also add them to method tests.